pkg/dbal/DbalConsumer.php

```diff
@@ -115,33 +97,54 @@ public function reject(Message $message, bool $requeue = false): void
 
             return;
         }
+
+        $this->deleteMessageByDeliveryId($message->getDeliveryId());
     }
 
-    protected function receiveMessage(): ?DbalMessage
+    private function receiveMessage(): ?DbalMessage
     {
+        $this->redeliverMessages();
+
         $this->dbal->beginTransaction();
         try {
-            $now = time();
-
-            $dbalMessage = $this->fetchPrioritizedMessage($now) ?: $dbalMessage = $this->fetchMessage($now);
-            if (false == $dbalMessage) {
+            $now = (int) time();
+            $redeliveryDelay = $this->getRedeliveryDelay() / 1000; // milliseconds to seconds
+            $deliveryId = Uuid::uuid1()->toString();
+
+            // get top message from the queue
+            $message = $this->fetchMessage($now);
```

Comment:
the method could return null, if no message in the queue. but you do not check that a few lines later and assume the message is there.